tools/convert.py

```diff
@@ -97,47 +135,50 @@ def handle_tensors(args, writer, state_dict):
             "img_in.",
             "guidance_in.",
             "final_layer.",
-        ]
-
-        if any([x in key for x in blacklist]) and ".weight" in key:
-            data_qtype = gguf.GGMLQuantizationType.F32
-
-        if n_dims == 1: 
-            # one-dimensional tensors should be kept in F32
-            # also speeds up inference due to not dequantizing
-            data_qtype = gguf.GGMLQuantizationType.F32
-        
-        elif n_params <= 1024:
-            # very small tensors
-            data_qtype = gguf.GGMLQuantizationType.F32
-        
-        elif n_dims == 4:
-            if min(data.shape[:2]) == 4: # output tensor
-                data_qtype = gguf.GGMLQuantizationType.F16
-            elif data_shape[-1] == 3: # 3x3 kernel
-                data_qtype = gguf.GGMLQuantizationType.F16
-            elif data_shape[-1] == 1: # 1x1 kernel
-                #data = np.squeeze(data) # don't do this
-                data_qtype = gguf.GGMLQuantizationType.F16
+        }
+
+        if old_dtype in (torch.float32, torch.bfloat16):
+            if n_dims == 1:
+                # one-dimensional tensors should be kept in F32
+                # also speeds up inference due to not dequantizing
+                data_qtype = gguf.GGMLQuantizationType.F32
+
+            elif n_params <= QUANTIZATION_THRESHOLD:
+                # very small tensors
+                data_qtype = gguf.GGMLQuantizationType.F32
+
+            elif ".weight" in key and any(x in key for x in blacklist):
+                data_qtype = gguf.GGMLQuantizationType.F32

```

Comment:
I think llama.cpp converts them to FP32 anyway during the quantization, which was the reason this was here. No one will use the FP16 checkpoints anyway just figured it's nicer if it matches since it barely increases the checkpoint size. We can leave them as fp16 though, it's fine.